src/query/storage/m3/storage.go

```diff
@@ -162,25 +170,22 @@ func (s *m3storage) FetchProm(
 		}
 
 		rollup := false
-		metricName := ""
+		nameTag := []byte{}
 		for _, label := range series.Labels {
 			// Check for both the rollup tag and the metric name label.
-			name := string(label.Name)
-			value := string(label.Value)
-			switch name {
-			case coordmodel.RollupTagName:
-				if value == coordmodel.RollupTagValue {
+			if bytes.Equal(label.Name, rollupTagName) {
+				if bytes.Equal(label.Value, rollupTagValue) {
 					rollup = true
 				}
-			case model.MetricNameLabel:
-				metricName = value
+			} else if bytes.Equal(label.Name, promDefaultName) {
+				nameTag = label.Value
 			}
 		}
 
 		if rollup {
-			fetchResult.Metadata.ByName(metricName).Aggregated++
+			fetchResult.Metadata.ByName(nameTag).Aggregated++
 		} else {
-			fetchResult.Metadata.ByName(metricName).Unaggregated++
+			fetchResult.Metadata.ByName(nameTag).Unaggregated++
```

Comment:
What do we do here if there's no `promDefaultName` tag? Is it valid to add to an unnamed fetch metadata?

Reply:
Yes adding to the "unnamed" metadata struct is valid. I will add an explicit test case for that.

Reply:
Test added.